### PR TITLE
skip parsing a declaration <!DOCTYPE>, but not get a syntax error

### PR DESCRIPTION
**Pull request by 7phs**
1. A XML document with a DOCTYPE declaration has parsed with a syntax error BadCommentOrCDATA

I add a code to parse <!DOCTYPE>. The parsing code is for checking syntax of DOCTYPE and just skipping it.

I want to make sure this is the right approach in the lexer.
1. I add a helper function to simplify check test cases with syntax error.

### CODE REVIEW

File: src/de/lexer.rs [hunk 1]
```diff
@@ -277,15 +277,16 @@ impl<Iter> XmlIterator<Iter>
     }
 
     fn decode_comment_or_cdata(&mut self) -> Result<(), LexerError> {
-        match try!(self.next_char()) {
+        match try!(self.peek_char()) {
             b'-' => self.decode_comment(),
             b'[' => self.decode_cdata(),
+            _ if self.ch==InternalLexical::StartOfFile => self.decode_prolog(),
```
Comment (oli-obk) [inline src/de/lexer.rs]: style nit: add spaces around the `==` operator

File: src/de/lexer.rs [hunk 2]
```diff
@@ -329,6 +330,101 @@ impl<Iter> XmlIterator<Iter>
...
+
+    fn decode_prolog_name(&mut self) -> Result<(), LexerError> {
+        self.buf.clear();
+
+        loop {
+            match try!(self.peek_char()) {
+                c if c.is_ws_or(b">") => {
+                    break
+                },
+                c if c == b'<' => {
```
Comment (oli-obk) [inline src/de/lexer.rs]: use `b'<'` as a pattern instead of comparing

Comment (7phs) [inline src/de/lexer.rs]: fixed

**Comment by oli-obk** [comment]
looks good to me. I like how the last 3 functions just munch away everything that's not needed. I'll publish a new version asap

**Comment by oli-obk** [comment]
published new version

**Pull request merged**